src/debug/debug-pal/win/diagnosticsipc.cpp

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <assert.h>
+#include <new>
+#include <stdio.h>
+#include "diagnosticsipc.h"
+
+IpcStream::DiagnosticsIpc::DiagnosticsIpc(const char(&namedPipeName)[MaxNamedPipeNameLength])
+{
+    memcpy(_pNamedPipeName, namedPipeName, sizeof(_pNamedPipeName));
```

Comment:
Can we use `memcpy_s` here?

Reply:
Here we are expecting an array by reference. Arrays of different size would generate a compile time error, as they are of different type.

Would you prefer if I change it to a C style array, pass the string length to be copied instead:
```diff
- IpcStream::DiagnosticsIpc::DiagnosticsIpc(const char(&namedPipeName)[MaxNamedPipeNameLength])
+ IpcStream::DiagnosticsIpc::DiagnosticsIpc(const char *const namedPipeName, const uint32_t namedPipeNameLength)
{
-     memcpy(_pNamedPipeName, namedPipeName, sizeof(_pNamedPipeName)); // Where the size of both arrays is the same.
+     memcpy_s(_pNamedPipeName, sizeof(_pNamedPipeName), namedPipeName, namedPipeNameLength);
}
```

On the other hand, I could check that the string is null terminated within the 256 characters allowed.

By the way, the length of 256 was chosen from this document: [lpName](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-createnamedpipea#parameters)
```
The unique pipe name. This string must have the following form:

\.\pipe<i>pipename

The pipename part of the name can include any character other than a backslash, including numbers and special characters. The entire pipe name string can be up to 256 characters long. Pipe names are not case sensitive.
```